go/src/koding/kites/kloud/stack/provider/variable.go

```diff
@@ -67,3 +81,40 @@ func ReplaceVariables(s string, vars []Variable, blank string) string {
 
 	return buf.String()
 }
+
+var escape = func(v *Variable) string {
+	return "$" + v.String()
+}
+
+// EscapeDeadVariables escapes variables which are commented out.
+//
+// The comment format is hardcoded to work for Bash-like shells.
+func EscapeDeadVariables(userdata string) string {
+	var buf bytes.Buffer
+
+	eofNL := strings.HasSuffix(userdata, "\n")
+	scanner := bufio.NewScanner(strings.NewReader(userdata))
+
+	for scanner.Scan() {
+		s := scanner.Text()
+
+		isComment := strings.HasPrefix(strings.TrimSpace(s), "#")
+
+		if !isComment {
+			fmt.Fprintln(&buf, s)
+			continue
+		}
+
+		fmt.Fprintln(&buf, ReplaceVariablesFunc(s, ReadVariables(s), escape))
+	}
+
+	// Scanning from strings.Reader is not going to fail, even if, there's
+	// no recovery from the failure.
+	_ = scanner.Err()
+
+	if eofNL {
+		return buf.String()
+	}
```

Comment:
here is something wrong with the logic. if it ends with new line, which is also true for the following case, we dont care about it and return the whole buffer, but if it doesnt have, we apply "TrimRight". In any case "Fprintln" will add new lines. What about removing this logic and using "TrimSpace"? Or am i missing something?